src/test/java/teammates/common/datatransfer/attributes/AccountRequestAttributesTest.java

```diff
@@ -15,89 +19,418 @@ public class AccountRequestAttributesTest extends BaseTestCase {
+        assertEquals(originalCreatedAt, accountRequestAttributes.getCreatedAt());
+        assertEquals(originalRegistrationKey, accountRequestAttributes.getRegistrationKey());
+        assertNull(accountRequestAttributes.getPureInstitute());
+        assertNull(accountRequestAttributes.getPureCountry());
+    }
+
+    @Test
+    public void testUpdateOptionsBuilder_withNullInput_shouldFailWithAssertionError() {
+        assertThrows(AssertionError.class, () -> AccountRequestAttributes.updateOptionsBuilder(null, "TMT, Singapore"));
+        assertThrows(AssertionError.class, () -> AccountRequestAttributes.updateOptionsBuilder("adam@tmt.tmt", null));
+        assertThrows(AssertionError.class, () ->
+                AccountRequestAttributes.updateOptionsBuilder("adam@tmt.tmt", "TMT, Singapore")
+                        .withName(null));
+        assertThrows(AssertionError.class, () ->
+                AccountRequestAttributes.updateOptionsBuilder("adam@tmt.tmt", "TMT, Singapore")
+                        .withInstitute(null));
+        assertThrows(AssertionError.class, () ->
+                AccountRequestAttributes.updateOptionsBuilder("adam@tmt.tmt", "TMT, Singapore")
+                        .withEmail(null));
+        assertThrows(AssertionError.class, () ->
+                AccountRequestAttributes.updateOptionsBuilder("adam@tmt.tmt", "TMT, Singapore")
+                        .withStatus(null));
+        assertThrows(AssertionError.class, () ->
+                AccountRequestAttributes.updateOptionsBuilder("adam@tmt.tmt", "TMT, Singapore")
+                        .withLastProcessedAt(null));
     }
 
     @Test
     public void testValidate() throws Exception {
+        ______TS("valid account request attributes, null pure institute and null pure country");
+
         AccountRequestAttributes validAccountRequest = getValidAccountRequestAttributesObject();
 
-        assertTrue("valid value", validAccountRequest.isValid());
+        assertTrue(validAccountRequest.isValid());
+
+        ______TS("valid account request attributes, non-null pure institute and pure country");
+
+        validAccountRequest = AccountRequestAttributes
+                .builder("Adam", "TMT", "Singapore", "adam@tmt.tmt", "valid url", "valid comments")
+                .build();
+
+        assertTrue(validAccountRequest.isValid());
+
+        ______TS("invalid account request attributes, null pure institute and null pure country");
 
-        String invalidEmail = "invalid-email";
         String emptyName = "";
+        String invalidEmail = "invalid-email";
+        String invalidInstitute = "{TMT, Singapore";
+        String longUrl = StringHelperExtension
+                .generateRandomAsciiStringOfLength(FieldValidator.ACCOUNT_REQUEST_HOME_PAGE_URL_MAX_LENGTH + 1);
+        String longComments = StringHelperExtension
+                .generateRandomAsciiStringOfLength(FieldValidator.ACCOUNT_REQUEST_COMMENTS_MAX_LENGTH + 1);
```

Comment:
These sort of tests shouldn't have random elements. We do not want different runs of the same test to give different results.

Reply:
This is a good suggestion. Let me consider whether I can change it to a fake random.

Reply:
Let me just use the existing one instead...